test/token/token.go

```diff
@@ -4,129 +4,151 @@ import (
-	var err error
-	token.Raw, err = token.SignedString(privateKey)
+	tokenStr, err := token.SignedString(key)
 	if err != nil {
-		return nil, errors.WithStack(err)
+		panic(errors.WithStack(err))
 	}
-	return token, nil
+	return tokenStr
 }
 
-// GenerateToken generates a JWT token and signs it using the given private key
-func GenerateToken(identityID string, identityUsername string, privateKey *rsa.PrivateKey) (string, error) {
-	token, err := GenerateTokenObject(identityID, identityUsername, privateKey)
-	if err != nil {
-		return "", errors.WithStack(err)
+func ContextWithRequest(ctx context.Context) context.Context {
+	if ctx == nil {
+		ctx = context.Background()
 	}
-	return token.Raw, nil
-}
-
-// NewManager returns a new token Manager for handling tokens
-func NewManager() token.Manager {
-	return token.NewManagerWithPublicKey("test-key", &PrivateKey().PublicKey)
-}
-
-// EmbedTokenInContext generates a token and embed it into the context
-func EmbedTokenInContext(t *testing.T, sub, username string) (context.Context, string, error) {
-	// Generate Token with an identity that doesn't exist in the database
-	tokenString, err := GenerateToken(sub, username, PrivateKey())
-	if err != nil {
-		return nil, "", err
+	u := &url.URL{
+		Scheme: "https",
+		Host:   "auth.openshift.io",
 	}
-
-	extracted, err := parse(t, tokenString)
+	rw := httptest.NewRecorder()
+	req, err := http.NewRequest("GET", u.String(), nil)
 	if err != nil {
-		return nil, "", err
-	}
-
-	// Embed Token in the context
-	ctx := jwtgoa.WithJWT(context.Background(), extracted)
-	ctx = tokencontext.ContextWithTokenManager(ctx, TokenManager)
-	return ctx, tokenString, nil
-}
-
-func parse(t *testing.T, tokenString string) (*jwt.Token, error) {
-	keyFunc := keyFunction()
-	jwtToken, err := jwt.Parse(tokenString, keyFunc)
-	require.NoError(t, err)
-	return jwtToken, nil
-}
-
-func keyFunction() jwt.Keyfunc {
-	return func(token *jwt.Token) (interface{}, error) {
-		kid := token.Header["kid"]
-		if kid == nil {
-			return nil, errors.New("there is no 'kid' header in the token")
-		}
-		key := TokenManager.PublicKey(fmt.Sprintf("%s", kid))
-		if key == nil {
-			return nil, errors.New(fmt.Sprintf("there is no public key with such ID: %s", kid))
-		}
-		return key, nil
+		panic(errors.WithStack(err))
 	}
+	return goa.NewContext(goa.WithAction(ctx, "Test"), rw, req, url.Values{})
 }
 
-func ContextWithTokenAndRequestID(t *testing.T) (context.Context, uuid.UUID, string, string) {
-	identityID := uuid.NewV4()
-	ctx, ctxToken, err := EmbedTokenInContext(t, identityID.String(), uuid.NewV4().String())
-	require.NoError(t, err)
-
+func ContextWithTokenAndRequestID() (context.Context, string, string) {
+	ctx, ctxToken := EmbedTokenInContext(uuid.NewV4().String(), uuid.NewV4().String())
+	ctx = tokencontext.ContextWithTokenManager(ctx, TokenManager)
 	reqID := uuid.NewV4().String()
 	ctx = client.SetContextRequestID(ctx, reqID)
 
-	return ctx, identityID, ctxToken, reqID
```

Comment:
why do you remove the `identityID` from the returned values? I believe it could be relevant in some test cases...

Reply:
I agree. Done in https://github.com/fabric8-services/fabric8-common/pull/27/commits/8ed151b61673a79673f9a29aebfeae4e360cf469

Reply:
thanks @alexeykazakov